CMakeLists.txt

```diff
@@ -35,14 +35,32 @@ endif()
 
 option(RUY_PROFILER "Enable ruy's built-in profiler (harms performance)" OFF)
 
+option(RUY_ENABLE_INSTALL "Enable install rule" ${RUY_IS_TOPLEVEL})
+
 include(cmake/ruy_add_all_subdirs.cmake)
 include(cmake/ruy_cc_library.cmake)
 include(cmake/ruy_cc_binary.cmake)
 include(cmake/ruy_cc_test.cmake)
 
+include(CMakeDependentOption)
+
+option(RUY_USE_EXTERNAL_CPUINFO "Use external cpuinfo" OFF)
+
+cmake_dependent_option(RUY_FIND_CPUINFO "Use find_package to find cpuinfo" ON "RUY_USE_EXTERNAL_CPUINFO" OFF)
+
 # Skip cpuinfo if it was already generated, which can happen when ruy is
 # a subdirectory in a wider project that already uses cpuinfo.
-if (NOT TARGET cpuinfo)
+if (RUY_USE_EXTERNAL_CPUINFO)
+  if (RUY_FIND_CPUINFO)
+    find_package(cpuinfo REQUIRED)
+  elseif (NOT TARGET cpuinfo::cpuinfo)
+    add_library(cpuinfo::cpuinfo ALIAS cpuinfo)
```

Comment:
That this `add_library` is made in 2 separate lines suggests at least asking: could this logic be simplified in some way so that this `add_library` would be needed only once?